### PR TITLE
python310Packages.tensorstore: init at 0.1.22

### PR DESCRIPTION
**Pull request by ethanabrooks**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/tensorstore/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+buildPythonPackage rec {
+  pname = "tensorstore";
+  version = "0.1.22";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-qQqytAtjXtThWh+a+dNLeQHlxYUGxdUOgK69aFDQtwA=";
+  };
+  buildInputs = [setuptools-scm];
+  propagatedBuildInputs = [numpy];
+  meta = with lib; {
```
Comment (onny) [inline pkgs/development/python-modules/tensorstore/default.nix:17]: Also check if you could ass `pythonImportsCheck`. Explicitly specifying `checkInputs = [ pytestCheckHook ];` is also recommended. If the tests are missing in the pypi release, you might have to fetch the sources from the GitHub repo

**Review by onny — COMMENTED** [review]
Thank you for your contribution. Just adding some minor comments :)

**Comment by ethanabrooks** [comment]
Hi @onny thank you for the input and sorry for the delayed response -- was traveling since Tuesday. I actually looked a little more deeply into `tensorstore` and this package is going to require a more sophisticated `default.nix` to package properly. I am happy to work on this, but may either need to brush up on my knowledge of `bazel` or receive guidance from someone who is more familiar with that build system.

File: pkgs/development/python-modules/tensorstore/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
```
[the remark this answers is not in the document]
Comment (ethanabrooks) [reply pkgs/development/python-modules/tensorstore/default.nix:1]: This is done.

File: pkgs/development/python-modules/tensorstore/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  setuptools-scm,
+}:
+buildPythonPackage rec {
+  pname = "tensorstore";
+  version = "0.1.22";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-qQqytAtjXtThWh+a+dNLeQHlxYUGxdUOgK69aFDQtwA=";
+  };
+  buildInputs = [setuptools-scm];
```
[the remark this answers is not in the document]
Comment (ethanabrooks) [reply pkgs/development/python-modules/tensorstore/default.nix:14]: I would have to get the full built to check. I got it from here.

File: pkgs/development/python-modules/tensorstore/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  propagatedBuildInputs = [numpy];
+  meta = with lib; {
+    homepage = "https://google.github.io/tensorstore/";
+    description = "TensorStore is a library for efficiently reading and writing large multi-dimensional arrays.";
+    license = licenses.asl20;
+    maintainers = with maintainers; [google_opensource jbms tensorstore];
```
[the remark this answers is not in the document]
Comment (ethanabrooks) [reply pkgs/development/python-modules/tensorstore/default.nix]: This is done.

**Comment by ethanabrooks** [comment]
> Before I forget: Please change commit message and the title of this PR to `python310Packages.tensorstore: init at X.X`

I replaced `X.X` with `0.1.22`, the package version number. I'm assuming that's what you wanted?

File: pkgs/development/python-modules/tensorstore/default.nix (hunk 1 above)
Comment (ethanabrooks) [reply pkgs/development/python-modules/tensorstore/default.nix:11]: I think I did this correctly, but please let me know.

**Comment by onny** [comment]
> > Before I forget: Please change commit message and the title of this PR to `python310Packages.tensorstore: init at X.X`
> 
> I replaced `X.X` with `0.1.22`, the package version number. I'm assuming that's what you wanted?

Thanks, thats good :)

**Comment by onny** [comment]
ofborg build feels to fails somehow

**Comment by tartavull** [comment]
What's missing to get this merged?

**Comment by collinarnett** [comment]
> What's missing to get this merged?

Looks like a maintainer needs to be added and the version could be bumped. I need this package to get orbax working to get flax unmarked as broken.

File: pkgs/development/python-modules/tensorstore/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  numpy,
+  setuptools-scm,
+}:
+buildPythonPackage rec {
+  buildInputs = [setuptools-scm];
+  checkInputs = [pytestCheckHook];
```
Comment (kirillrdy) [inline pkgs/development/python-modules/tensorstore/default.nix:10]: `pytestCheckHook` needs to be added to list of arguments to this file

File: pkgs/development/python-modules/tensorstore/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  meta = with lib; {
```
Comment (kirillrdy) [inline pkgs/development/python-modules/tensorstore/default.nix:11]: usually `meta` is at the bottom of the file ( I haven't seen a case when it isn't )

**Comment by ethanabrooks** [comment]
Moved `meta` to the bottom.

File: pkgs/development/python-modules/tensorstore/default.nix (hunk 5 above)
Comment (kirillrdy) [reply pkgs/development/python-modules/tensorstore/default.nix:10]: still an issue, fails to build because of this

**Comment by kirillrdy** [comment]
superseded by https://github.com/NixOS/nixpkgs/pull/245792

**Pull request closed without merging**